python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/datahub_client.py

```diff
@@ -146,6 +176,212 @@ def create_domain(
+            aspect=DataPlatformInstance(
+                platform="urn:li:dataPlatform:athena",
+            ),
+        )
+        self.graph.emit(metadata_event)
+        logger.info(f"Platform updated for Database {name} ")
+
+        # container type update
+        metadata_event = MetadataChangeProposalWrapper(
+            entityUrn=database_urn,
+            aspect=SubTypesClass(typeNames=[DatasetContainerSubTypes.DATABASE]),
+        )
+        self.graph.emit(metadata_event)
+        logger.info(f"Type updated for Database {name} ")
+
+        return database_urn
+
+    def upsert_athena_table(
+        self,
+        metadata: TableMetadata,
+        database_metadata: DatabaseMetadata | None = None,
+    ) -> str:
+        if not metadata.parent_database_name:
+            raise ValueError("parent_database_name needs to be set in TableMetadata")
+
+        fully_qualified_name = f"{metadata.parent_database_name}.{metadata.name}"
+
+        dataset_urn = mce_builder.make_dataset_urn(
+            platform="athena", name=fully_qualified_name, env="PROD"
+        )
+        # jscpd:ignore-start
+        dataset_properties = DatasetPropertiesClass(
+            name=metadata.name,
+            qualifiedName=fully_qualified_name,
+            description=metadata.description,
+            customProperties={
+                "sourceDatasetName": metadata.source_dataset_name,
+                "whereToAccessDataset": metadata.where_to_access_dataset,
+                "sensitivityLevel": metadata.data_sensitivity_level.name,
+                **(
+                    {"rowCount": str(metadata.row_count)}
+                    if metadata.row_count is not None
+                    else {}
+                ),
+            },
+        )
+
+        metadata_event = MetadataChangeProposalWrapper(
+            entityUrn=dataset_urn,
+            aspect=dataset_properties,
+        )
+        self.graph.emit(metadata_event)
+
+        dataset_schema_properties = SchemaMetadataClass(
+            schemaName=metadata.name,
+            platform="urn:li:dataPlatform:athena",
+            version=metadata.major_version,
+            hash="",
+            platformSchema=OtherSchemaClass(rawSchema=""),
+            fields=[
+                SchemaFieldClass(
+                    fieldPath=f"{column['name']}",
+                    type=SchemaFieldDataTypeClass(
+                        type=DATAHUB_DATA_TYPE_MAPPING[column["type"]]
+                    ),
+                    nativeDataType=column["type"],
+                    description=column["description"],
+                )
+                for column in metadata.column_details
+            ],
+        )
+
+        metadata_event = MetadataChangeProposalWrapper(
+            entityType="dataset",
+            changeType=ChangeTypeClass.UPSERT,
+            entityUrn=dataset_urn,
+            aspect=dataset_schema_properties,
+        )
+        self.graph.emit(metadata_event)
```

Comment:
If we start building more ingestion libraries, we can perhaps lift and shift this code